docs/developer-docs/security/general-security-best-practices.md

```diff
@@ -107,3 +107,25 @@ For example, we have seen issues where the public key to verify certification wa
 #### Recommendation
 
 Avoid test and dev code in production code whenever possible.
+
+### Data confidentiality on the Internet Computer
```

Comment:
Thanks Robin, I had overlooked that heading. I've moved this section up above that heading now. 

I think this section fits best in this 'General best practices' page since it is more of a general piece of information about what developers can expect regarding default data confidentiality rather than a direct recommendation for canister data encryption.